app/upgrades/v11/upgrades.go

```diff
@@ -58,3 +78,80 @@ func MigrateEscrowAccounts(ctx sdk.Context, ak authkeeper.AccountKeeper) {
 		ak.SetModuleAccount(ctx, acc)
 	}
 }
+
+// HandleMainnetUpgrade handles the logic for the reward distribution, it only commits to the db if successful
+func HandleRewardDistribution(ctx sdk.Context, bk bankkeeper.Keeper, sk stakingkeeper.Keeper, dk distributionkeeper.Keeper, logger log.Logger) {
+	// use a cache context as a rollback mechanism in case
+	// the distrbution fails
+	cacheCtx, writeFn := ctx.CacheContext()
+	err := DistributeRewards(cacheCtx, bk, sk, dk)
+	if err != nil {
+		// log error instead of aborting the upgrade
+		logger.Error("failed to distribute rewards", "error", err.Error())
+	} else {
+		writeFn()
+	}
+}
+
+// DistributeRewards distributes the token allocations from the Olympus Mons incentivized testnet
+func DistributeRewards(ctx sdk.Context, bk bankkeeper.Keeper, sk stakingkeeper.Keeper, dk distributionkeeper.Keeper) error {
+	fundingAccountAddress := sdk.MustAccAddressFromBech32(FundingAccount)
+	numValidators := sdk.NewInt(int64(len(Validators)))
+
+	for _, currentDistribute := range Accounts {
+
+		// move rewards to the receiving account
+		receivingAccount := sdk.MustAccAddressFromBech32(currentDistribute[0])
+		receivingAmount, ok := sdk.NewIntFromString(currentDistribute[1])
+		if !ok {
+			return errorsmod.Wrapf(errortypes.ErrInvalidType,
+				"cannot retrieve allocation from string for address %s",
+				currentDistribute[0])
+		}
+		currentRewards := sdk.Coins{
```

Comment:
not sure, I understand the meaning of `current` here. Keep it simple
```suggestion
		reward := sdk.Coins{
```